Extensions/DialogueTree/JsExtension.js

```diff
@@ -0,0 +1,512 @@
+				_('Scroll clipped text'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.getCodeExtraInformation()
+			.setFunctionName('gdjs.dialogueTree.scrollClippedText');
+
+		extension
+			.addAction(
+				'SetVariable',
+				_('Set dialogue state variable'),
+				_('Set dialogue state variable. Use this to set a variable that the dialogue data is using.'),
+				_('Set dialogue state variable _PARAM0_ to _PARAM1_'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.addParameter('string', _('State Variable Name'))
+			.addParameter('expression', _('Variable Value'))
+			.getCodeExtraInformation()
+			.setFunctionName('gdjs.dialogueTree.setVariable');
+
+		extension
+			.addAction(
+				'SaveState',
+				_('Save dialogue state'),
+				_('Save dialogue state. Use this to store the dialogue state into a variable, which can later be used for saving the game. That way player choices can become part of the game save.'),
+				_('Save dialogue state to _PARAM0_'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.addParameter('globalvar', _('Global Variable'))
+			.getCodeExtraInformation()
+			.setFunctionName('gdjs.dialogueTree.saveState');
+
+		extension
+			.addAction(
+				'LoadState',
+				_('Load dialogue state'),
+				_('Load dialogue state. Use this to restore dialogue state, if you have stored in a variable before with the "Save state" action.'),
+				_('Load dialogue state from _PARAM0_'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.addParameter('globalvar', _('Global Variable'))
+			.getCodeExtraInformation()
+			.setFunctionName('gdjs.dialogueTree.loadState');
+
+		extension
+			.addStrExpression(
+				'LineText',
+				_('Get the current dialogue line text'),
+				_('Returns the current dialogue line text'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.getCodeExtraInformation()
+			.setFunctionName('gdjs.dialogueTree.getLineText');
+
+		extension
+			.addExpression(
+				'OptionsCount',
+				_('Get the number of options in an options line type'),
+				_('Get the number of options in an options line type'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.getCodeExtraInformation()
+			.setFunctionName('gdjs.dialogueTree.getLineOptionsCount');
+
+		extension
+			.addStrExpression(
+				'Option',
+				_('Get the text of an option from an Options line type'),
+				_('Get the text of an option from an Options line type, using the option\'s Number. The numbers start from 0.),
```

Comment:
You have a syntax error, double check also the rest of the extension

Reply:
oops, sorry. Somehow managed to wipe the closing '